libevm/lib/service_state.go

```diff
@@ -112,6 +116,14 @@ func (s *Service) StateCommit(params HandleParams) (error, common.Hash) {
 	return nil, hash
 }
 
+// TODO: we might want to change this to "StateEmpty()" because the logic behind Exists() is a bit unpredictable
```

Comment:
good, let's do it and change in the places of usage.

Reply:
done